src/common/status.h

```diff
@@ -105,11 +105,61 @@ class [[nodiscard]] Status {
   friend struct StatusOr;
 };
 
+namespace type_details {
 template <typename... Ts>
-using first_element = typename std::tuple_element<0, std::tuple<Ts...>>::type;
+using first_element_t = typename std::tuple_element_t<0, std::tuple<Ts...>>;
 
 template <typename T>
-using remove_cvref_t = typename std::remove_cv<typename std::remove_reference<T>::type>::type;
+using remove_cvref_t = typename std::remove_cv_t<typename std::remove_reference_t<T>>;
+};  // namespace type_details
+
+template <typename, typename = void>
+struct StringInStatusOr : private std::string {
+  using base_type = std::string;
+  static constexpr bool inplace = true;
+
+  explicit StringInStatusOr(std::string&& v) : base_type(std::move(v)) {}
```

Comment:
Across the `Kvrocks` codebase, we place `&` and `*` near the value instead of near the type. So in order to keep the same code style, you can place them close to variables or the `operator` keyword. However, if there is an exception for templates or clang-format tool forces such syntax, it's OK.

Reply:
I tried to change it to `std::string &&v`, but clang-format changed it back.